### PR TITLE
PR CL-104 AddIntegrationTestForLeadController

### PR DESCRIPTION
**Pull request by OleksandraKondratenko**

### CODE REVIEW

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public class LeadControllerIntegrationTest : IntegrationTestAbstract
+    {
+        public LeadControllerIntegrationTest(ApiWebApplicationFactory fixture) : base(fixture)
+        {
+
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetAll), MemberType = typeof(LeadTestData))]
+        public async Task GetAllLeads_WhenValidTestPassed_ShouldReturnIEnumerableWallets(List<Lead> leads)
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: ShouldReturnIEnumerableWallets?

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            var getRoute = $"/api/Lead";
+            var getResponse = await SendRequestToGetAll(getRoute);
+            var actual = JsonConvert.DeserializeObject<IEnumerable<Lead>>(await getResponse.Content.ReadAsStringAsync());
+
+            actual.Should().BeEquivalentTo(leads);
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetOne), MemberType = typeof(LeadTestData))]
+        public async Task GetLeadByID_WhenValidTestPassed_ShouldReturnIEnumerableWallets(Lead lead)
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: ShouldReturnIEnumerableWallets?

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            var getRoute = $"/api/Lead/{lead.Id}";
+            var getResponse = await SendRequestToGetByID(getRoute);
+            var actual = JsonConvert.DeserializeObject<Lead>(await getResponse.Content.ReadAsStringAsync());
+
+            actual.Should().BeEquivalentTo(lead);
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetOne), MemberType = typeof(LeadTestData))]
+        public async Task UpdateLead_WhenValidTestPassed_ShouldReturnIEnumerableWallets(Lead lead)
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: ShouldReturnIEnumerableWallets?

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            
+            var getRoute = $"/api/Lead/{lead.Id}";
+            var getResponse = await SendRequestToGetByID(getRoute);
+            var actual = JsonConvert.DeserializeObject<Lead>(await getResponse.Content.ReadAsStringAsync());
+            actual.Should().BeEquivalentTo(lead);
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetOne), MemberType = typeof(LeadTestData))]
+        public async Task DeleteLead_WhenValidTestPassed_ShouldReturnIEnumerableWallets(Lead lead)
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: ShouldReturnIEnumerableWallets?

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+            var getRoute = $"/api/Lead";
+            var getResponse = await SendRequestToGetAll(getRoute);
+            var actual = JsonConvert.DeserializeObject<IEnumerable<Lead>>(await getResponse.Content.ReadAsStringAsync());
+
+            actual.Should().BeEquivalentTo(leads);
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetOne), MemberType = typeof(LeadTestData))]
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: ForGetByID?

File: CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            var putResponse = await SendRequestToUpdate(lead,putRoute);
+            
+            var getRoute = $"/api/Lead/{lead.Id}";
+            var getResponse = await SendRequestToGetByID(getRoute);
+            var actual = JsonConvert.DeserializeObject<Lead>(await getResponse.Content.ReadAsStringAsync());
+            actual.Should().BeEquivalentTo(lead);
+        }
+
+        [Theory]
+        [MemberData(nameof(LeadTestData.GetTestDataForGetOne), MemberType = typeof(LeadTestData))]
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/ControllerTests/LeadControllerIntegrationTest.cs]: GetByID?

File: CRMLite.CRM.IntegrationTests/SourceData/TestSourceData/LeadTestData.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                Id = Guid.NewGuid(),
+                FirstName = "Ola",
+                LastName = "Teranovich",
+                Email = "oleksandrakondratenko@gmail.com",
+                PassportNumber = "AE123456",
+                Password = "Aa12345!",
+                Role =null,
+                StatusType = StatusType.Regular,
+                TIN="123456791"
```
Comment (SemBen) [inline CRMLite.CRM.IntegrationTests/SourceData/TestSourceData/LeadTestData.cs]: Ctr+K+D

File: CRMLite.Core/Contracts/Authentification/StarupExrension.cs [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
 namespace CRMLite.Core.Contracts.Authentification
 {
-    public static class CreateToken
+    public static class StarupExrension
```
Comment (SemBen) [inline CRMLite.Core/Contracts/Authentification/StarupExrension.cs:8]: StartupExtension

File: CRMLite.Core/Contracts/Authorization/Roles/StatusType.cs [hunk 9]
```diff
@@ -2,7 +2,7 @@
 {
     public enum StatusType 
     {
-        Regular=0,
```
Comment (SemBen) [inline CRMLite.Core/Contracts/Authorization/Roles/StatusType.cs:5]: молодец! не забыла)

**Review by SemBen — COMMENTED** [review]
Исправь

File: CRMLite.Core/Contracts/Authentification/StarupExrension.cs (hunk 8 above)
Comment (Aryewar) [inline CRMLite.Core/Contracts/Authentification/StarupExrension.cs:8]: Mistake in word  Exrension(StartupExtension)

**Review by Aryewar — APPROVED** [review]

**Review by SemBen — APPROVED** [review]
ОК

**Pull request merged**